File: front/lib/models/file_analysis_result_dto.dart
class FileAnalysisResultDTO {
  final int id;
  final String promptName;
  final String compliance;
  final String issues;
  final String recommendations;

  FileAnalysisResultDTO({
    required this.id,
    required this.promptName,
    required this.compliance,
    required this.issues,
    required this.recommendations,
  });

  factory FileAnalysisResultDTO.fromJson(Map<String, dynamic> json) {
    return FileAnalysisResultDTO(
      id: json['id'],
      promptName: json['prompt_name'],
      compliance: json['compliance'],
      issues: json['issues'],
      recommendations: json['recommendations'],
    );
  }
}

File: api/internal/dto/file_analysis_result_dto.go
package dto

type FileAnalysisResultDTO struct {
	ID              uint   `json:"id"`
	PromptName      string `json:"prompt_name"`
	Compliance      string `json:"compliance"`
	Issues          string `json:"issues"`
	Recommendations string `json:"recommendations"`
}

type GetFileAnalysisResultsResponse struct {
	FileAnalysisResults []FileAnalysisResultDTO `json:"analysis_results"`
}
